wpilibc/src/main/native/cpp/Commands/ConditionalCommand.cpp

```diff
@@ -48,48 +35,36 @@ ConditionalCommand::ConditionalCommand(Command* onTrue, Command* onFalse) {
  */
 ConditionalCommand::ConditionalCommand(const std::string& name, Command* onTrue,
                                        Command* onFalse)
-    : Command(name) {
-  m_onTrue = onTrue;
-  m_onFalse = onFalse;
-
-  RequireAll(*this, onTrue, onFalse);
+    : CommandGroup(name) {
+  m_onTrue = processCommand(onTrue);
+  m_onFalse = processCommand(onFalse);
 }
 
 void ConditionalCommand::_Initialize() {
+  CommandGroup::_Initialize();
+
   if (Condition()) {
-    m_chosenCommand = m_onTrue;
+    SetCommands(m_onTrue);
   } else {
-    m_chosenCommand = m_onFalse;
-  }
-
-  if (m_chosenCommand != nullptr) {
-    /*
-     * This is a hack to make cancelling the chosen command inside a
-     * CommandGroup work properly
-     */
-    m_chosenCommand->ClearRequirements();
-
-    m_chosenCommand->Start();
+    SetCommands(m_onFalse);
   }
 }
 
-void ConditionalCommand::_Cancel() {
-  if (m_chosenCommand != nullptr && m_chosenCommand->IsRunning()) {
-    m_chosenCommand->Cancel();
+std::vector<CommandGroupEntry> ConditionalCommand::processCommand(
+    Command* cmd) {
+  std::vector<CommandGroupEntry> list;
+  if (cmd == nullptr) {
+    return list;
```

Comment:
This returns an empty vector. When the CommandGroup is started, it will see that its index is equal to the length of the vector and do nothing during `_Execute`, then return true in `IsFinished`.

We have tested having an empty list in Python, but can't test the C++ version because we don't have a C++ program for our robot and allwpilib fails to build on Arch Linux with linker errors. We get the same error from the [arm-frc-linux-gnueabi-wpilib-git](https://aur.archlinux.org/packages/arm-frc-linux-gnueabi-wpilib-git/) AUR package.

Reply:
Which linker errors? I'm able to build a simple robot program after linking against libwpi.so and pthread. CTRE's libraries don't link though.

Reply:
What compiler are you using? For 2018, we are moving to gcc 5.4, and all dependencies have been built with that version. If you are still using last years compiler, you'll need to recompile the dependencies as well or else you'll get linker errors.

Reply:
The one in the AUR is 4.9. I have a beta toolchain at the following mirror: https://file.tavsys.net/archlinux/frc-beta/